snowplow-demo-app/src/androidTest/java/com/snowplowanalytics/snowplowtrackerdemo/uitest/UITest.java

```diff
@@ -0,0 +1,179 @@
+import static androidx.test.espresso.action.ViewActions.replaceText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static com.snowplowanalytics.snowplow.tracker.constants.TrackerConstants.SCHEMA_APPLICATION;
+import static com.snowplowanalytics.snowplow.tracker.constants.TrackerConstants.SCHEMA_GDPR;
+import static com.snowplowanalytics.snowplowtrackerdemo.BuildConfig.MICRO_ENDPOINT;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@Ignore("Automation isn't completed yet, comment this line when it is run locally, see #328 at github repo.")
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class UITest {
+
+    private String micro_uri;
+    private String micro_all_url;
+    private String micro_good_url;
+    private String micro_bad_url;
+    private String micro_reset_url;
+
+    private OkHttpClient client;
+    private Request resetRequest;
+
+    private JsonParser parser = new JsonParser();
+
+    @Rule
+    public ActivityTestRule<Demo> activityRule
+            = new ActivityTestRule<>(Demo.class);
+
+    @Before
+    public void beforeAll() {
+
+        // unlock screen
+        final Demo activity = activityRule.getActivity();
+        Runnable wakeUpDevice = new Runnable() {
+            public void run() {
+                activity.getWindow().addFlags(
+                    WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON
+                    | WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED
+                    | WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+            }
+        };
+        activity.runOnUiThread(wakeUpDevice);
+
+        // set micro endpoints
+        micro_uri = MICRO_ENDPOINT;
+        micro_all_url = "https://".concat(MICRO_ENDPOINT.concat("/micro/all"));
+        micro_good_url = "https://".concat(MICRO_ENDPOINT.concat("/micro/good"));
+        micro_bad_url = "https://".concat(MICRO_ENDPOINT.concat("/micro/bad"));
+        micro_reset_url = "https://".concat(MICRO_ENDPOINT.concat("/micro/reset"));
+
+        // init okhttp client
+        client = new OkHttpClient();
+
+        resetRequest = new Request.Builder()
+                .url(micro_reset_url)
+                .build();
+    }
+
+    @Test
+    public void t1_sendDemoEvents() throws InterruptedException, IOException, JSONException {
+        client.newCall(resetRequest).execute();
+
+        Espresso.closeSoftKeyboard();
+        onView(withId(R.id.emitter_uri_field)).perform(replaceText(micro_uri), closeSoftKeyboard());
+        onView(withId(R.id.emitter_uri_field)).check(matches(withText(micro_uri)));
+        onView(withId(R.id.btn_lite_start)).perform(click());
+
+        // TODO this is not best practice for idling resources
+        Thread.sleep(20000);
```

Comment:
This seems excessive and is introducing 20 seconds to a test suite!  This should really be refactored to not require any sleeps.

Reply:
It was based on demo app, shorter sleeps can end the tests without letting the app send all events.

I'm aware of the disadvantage and I was planning to change the approach later since this isn't a proper way. Although test duration won't change much, I'll change the approach right in this release.

Reply:
I think for this kind of thing its better not to leave it!  Let me know if you need any help solving it.